src/components/KTable/KTable.vue

```diff
@@ -1040,6 +1066,18 @@ const getTestIdString = (message: string): string => {
   return msg
 }
 
+watch([columnVisibility, tableHeaders], (newVals) => {
+  const newVisibility = newVals[0]
+  const newHeaders = newVals[1]
+  const newVisibleHeaders = newHeaders.filter((header: TableHeader) => newVisibility[header.label] !== false)
+
+  if (JSON.stringify(newVisibleHeaders) !== JSON.stringify(visibleHeaders.value)) {
+    emitTablePreferences()
+  }
+
+  visibleHeaders.value = newVisibleHeaders
```

Comment:
question: should we set the value before emitting?

Also, if the objects are equal, do we still need to set the `visibleHeaders.value` again?